rtengine/dcraw.cc

```diff
@@ -10007,10 +10009,23 @@ static inline uint32_t DNG_FP24ToFloat(const uint8_t * input) {
 static void expandFloats(Bytef * dst, int tileWidth, int bytesps) {
   if (bytesps == 2) {
     uint16_t * dst16 = (uint16_t *) dst;
+#ifndef __F16C__
     uint32_t * dst32 = (uint32_t *) dst;
     for (int index = tileWidth - 1; index >= 0; --index) {
-      dst32[index] = DNG_HalfToFloat(dst16[index]);
+        dst32[index] = DNG_HalfToFloat(dst16[index]);
+    }
+#else
+    float * dst32 = (float *) dst;
+    int index = tileWidth - 4;
+    for (; index >= 0; index -= 4) {
```

Comment:
If I remember correctly, you once told me, that pre-caching works better, if you walk the cache line forward instead of backward. Shouldn't that be applied here, too?

Reply:
I thought that too, but it won't work here because src and dst are the same buffer.